apps/emqx_bridge_s3/src/emqx_bridge_s3_aggreg_csv.erl

```diff
@@ -0,0 +1,97 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2022-2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+
+%% CSV container implementation for `emqx_bridge_s3_aggregator`.
+-module(emqx_bridge_s3_aggreg_csv).
+
+%% Container API
+-export([
+    new/1,
+    fill/2,
+    close/1
+]).
+
+-export_type([container/0]).
+
+-record(csv, {
+    columns :: [binary()] | undefined,
+    order :: [binary()],
+    separator :: char() | iodata(),
+    delimiter :: char() | iodata(),
```

Comment:
Nit: m.b. record_separator/column_separator and line_separator?

Reply:
Good point. Turns out there's tiny difference in semantics between _separator_ and _delimiter_, so I just put appropriate prefixes for clarity.